File: exp-04/q2-all-pairs-with-given-sum/Solution.java
import java.util.Scanner;

public class Solution {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] a = new int[n];
        for (int i = 0; i < n; i++)
            a[i] = sc.nextInt();
        int s = sc.nextInt();
        boolean found = false;
        // Brute force: try every pair (i, j) with i < j
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                if (a[i] + a[j] == s) {
                    System.out.println("(" + i + ", " + j + ")");
                    found = true;
                }
            }
        }
        if (!found)
            System.out.println("No pairs found");
        sc.close();
    }
}

File: exp-04/q2-all-pairs-with-given-sum/solution.cpp
#include <iostream>
using namespace std;

int main()
{
    int n, s;
    cin >> n;
    int a[n];
    for (int i = 0; i < n; i++)
        cin >> a[i];
    cin >> s;
    bool found = false;
    // Brute force: try every pair (i, j) with i < j
    for (int i = 0; i < n; i++)
    {
        for (int j = i + 1; j < n; j++)
        {
            if (a[i] + a[j] == s)
            {
                cout << "(" << i << ", " << j << ")" << endl;
                found = true;
            }
        }
    }
    if (!found)
        cout << "No pairs found" << endl;
    return 0;
}